selfdrive/boardd/panda.h

```diff
@@ -113,4 +114,11 @@ class Panda {
   uint8_t len_to_dlc(uint8_t len);
   void can_send(capnp::List<cereal::CanData>::Reader can_data_list);
   bool can_receive(std::vector<can_frame>& out_vec);
+
+protected:
+  // for unit tests
+  Panda(uint32_t bus_offset) : bus_offset(bus_offset) {}
```

Comment:
why can't these be moved to the test class?

Reply:
because the bus_offset is const, it can only be initialized in base class's CTOR